pkg/vsock/handshake_windows.go

```diff
@@ -48,76 +49,73 @@ func GetVMGUID(seed string, handshakePort uint32, timeout <-chan time.Time) (hvs
 
 	key.Close()
 
+	ctx, cancel := context.WithCancel(context.Background())
 	found := make(chan hvsock.GUID, len(names))
-	done := make(chan bool, len(names))
-	defer close(done)
 
 	for _, name := range names {
 		vmGUID, err := hvsock.GUIDFromString(name)
 		if err != nil {
 			logrus.Errorf("invalid VM name: [%s], err: %v", name, err)
 			continue
 		}
-		go handshake(vmGUID, seed, handshakePort, found, done)
+		go handshake(ctx, vmGUID, handshakePort, seed, found)
 	}
-	return tryFindGUID(found, timeout)
+	return tryFindGUID(cancel, found, timeout)
 }
 
-// handshake attempts to perform a handshake by verifying the seed with a running
-// af_vsock peer in WSL distro, it attempts once per second
-func handshake(vmGUID hvsock.GUID, seedPhrase string, peerHandshakePort uint32, found chan<- hvsock.GUID, done <-chan bool) {
-	svcPort, err := hvsock.GUIDFromString(winio.VsockServiceID(peerHandshakePort).String())
+// GetVsockConnection estabilishes a new AF_VSOCK connection with
+// the provided VM GUID and port, the caller is responsible for closing the connection
+func GetVsockConnection(vmGUID hvsock.GUID, port uint32) (net.Conn, error) {
+	svcPort, err := hvsock.GUIDFromString(winio.VsockServiceID(port).String())
 	if err != nil {
-		logrus.Errorf("hostHandshake parsing svc port: %v", err)
+		return nil, err
 	}
 	addr := hvsock.Addr{
 		VMID:      vmGUID,
 		ServiceID: svcPort,
 	}
 
+	return hvsock.Dial(addr)
+}
+
+// handshake attempts to perform a handshake by verifying the seed with a running
+// af_vsock peer in WSL distro, it attempts once per second
```

Comment:
Grammar fixes:
```suggestion
// handshake with the Hyper-V VM by verifying the fixed seed over AF_VSOCK once
// per second, in order to identify the VM running the WSL distro.
```
I'm not sure "seed" is the correct term here; it's closer to a secret or signature?